File: src/main/java/com/restfeel/config/security/SecurityConfig.java
///*
// * Copyright 2014 Ranjan Kumar
// *
// * Licensed under the Apache License, Version 2.0 (the "License");
// * you may not use this file except in compliance with the License.
// * You may obtain a copy of the License at
// *
// *      http://www.apache.org/licenses/LICENSE-2.0
// *
// * Unless required by applicable law or agreed to in writing, software
// * distributed under the License is distributed on an "AS IS" BASIS,
// * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// * See the License for the specific language governing permissions and
// * limitations under the License.
// */
//package com.restfeel.config.security;
//
//import com.restfeel.security.RfUserDetailService;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.context.annotation.Bean;
//import org.springframework.context.annotation.Configuration;
//import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
//import org.springframework.security.config.annotation.web.builders.HttpSecurity;
//import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
//import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
//import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
//import org.springframework.security.crypto.password.PasswordEncoder;
//
//@Configuration
//@EnableWebSecurity
//public class SecurityConfig extends WebSecurityConfigurerAdapter {
//
//    @Autowired
//    private RfUserDetailService rfUserDetailService;
//
//    @Autowired
//    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
//        auth.userDetailsService(rfUserDetailService).passwordEncoder(this.passwordEncoder());
//    }
//
//    @Bean
//    public PasswordEncoder passwordEncoder() {
//        return new BCryptPasswordEncoder();
//    }
//
//    @Override
//    protected void configure(HttpSecurity http) throws Exception {
//        http.csrf().disable();
//
//        http.authorizeRequests()
//            .antMatchers("/api/**", "/about", "/fonts/**", "/css/**", "/img/**", "/js/**", "/less/**",
//                "/scss/**")//默认不拦截静态资源等url pattern
//            .permitAll()
//            .anyRequest()
//            .authenticated()
//            .and().formLogin().loginPage("/login")
//            .failureUrl("/login?error").permitAll();
//
//        http.logout().logoutSuccessUrl("/");
//    }
//}

File: src/main/kotlin/com/restfeel/config/SecurityConfig.kt
package com.restfeel.config

import com.restfeel.security.RfUserDetailService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

/**
 * Created by jack on 2017/3/29.
 */


@Configuration
@EnableWebSecurity
class SecurityConfig : WebSecurityConfigurerAdapter() {
    @Autowired
    private val rfUserDetailService: RfUserDetailService? = null

    @Autowired
    @Throws(Exception::class)
    fun configureGlobal(auth: AuthenticationManagerBuilder) {
        auth.userDetailsService<RfUserDetailService>(rfUserDetailService).passwordEncoder(this.passwordEncoder())
    }

    @Bean
    fun passwordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }

    @Throws(Exception::class)
    override fun configure(http: HttpSecurity) {
        http.csrf().disable()

        http.authorizeRequests()
                .antMatchers("/api/**", "/about", "/fonts/**", "/css/**", "/img/**", "/js/**", "/less/**",
                        "/scss/**")//默认不拦截静态资源等url pattern
                .permitAll()
                .anyRequest()
                .authenticated()
                .and().formLogin().loginPage("/login")
                .failureUrl("/login?error").permitAll()

        http.logout().logoutSuccessUrl("/")
    }

}
